clang/lib/Driver/Driver.cpp

```diff
@@ -8457,9 +8457,17 @@ class ToolSelector final {
     return dyn_cast<JobAction>(CurAction);
   }
 
+  // We need to collapse the separate compilation steps when performing
+  // a third party host compilation step for SYCL offloading.  We don't know
```

Comment:
What happens when we do host compilations using Clang instead of third party compilers like g++?

Reply:
We behave pretty much the same as we do with `g++` as the third party compiler.

Reply:
nit: Ok. Since the comment mentions only third-party compilers, it is not clear if Clang supports -save-temps or not.

Reply:
The support of `clang` and `-save-temps` is inconsequential here.  The `-save-temps` support behavior we are modifying is explicit to our compilation flow and has nothing to do with the third party compiler.  We use the third party compiler to create the host object, we do not expect that compiler to do anything more than that.

Reply:
So, I was trying to understand what the behavior would be when we do something like `-fsycl-host-compiler=clang -fsycl-host-compiler-options=<clang options>` `-save-temps -fsycl.` ( since clang is not a 3rd party compiler)
But I looked at the documentation and it appears that when we use fsycl-host-compiler it implicitly means 3rd party compiler like `g+= or cl`. So maybe `fsycl-host-compiler=clang` is in itself invalid invocation.

Reply:
It can be any third party compiler you want to use - we call out `g++` and `cl` as they are common uses.  There is no restriction on using `clang` since it can be a different version of `clang` the user has on their system that they want to use and not use what we provide.